### ISSUE TITLE
 [video_player] Sometimes the plugin doesn't work

### ISSUE THREAD

**Issue by rubenteste** [flutter/flutter#81480]
I am using this plugin together with the arkit plugin, using one after the other. As soon as I detect 1 image with the arkit I show the video using the video plugin. But I am having problems because sometimes the video is not shown, or everything is white, or the video only shows the first frame and shows nothing more and sometimes it works.

Does not show any error and only pass the time that the video should show and then closes, because I'm using _PlayerVideoAndPopPageState. I even use future delay with some seconds and it's the same thing.

How can I fix this?

**Comment by TahaTesser** [comment]
Hi @rubenteste 
It seems you've asked a similar question before https://github.com/flutter/flutter/issues/78986

If you're experiencing an issue with `video_player` itself,   please provide your `flutter doctor -v`, your `pubspec.yaml`, a minimal reproducible code sample that does not use 3rd party plugins, your complete `flutter run -v` logs running the minimal code sample 
Thank you

**Comment by TahaTesser** [comment]
Hi @rubenteste 

> If you're experiencing an issue with `video_player` itself,   please provide your `flutter doctor -v`, your `pubspec.yaml`, a minimal reproducible code sample that does not use 3rd party plugins, your complete `flutter run -v` logs running the minimal code sample 

As requested please provide a reproducible code sample using `video_player` without using third-party packages such as `arkit_plugin`.
Thank you

**Comment by rubenteste** [comment]
HI,
I don't understand why I can't send code that contains the arkit plugin ... I would say that the bug is really from the video plugin ... and I can't quite reproduce the problem that happens because the arkit uses a lot of the phone's resources and when the plugin start video is very likely not working very well with the arkit which probably makes him want to use features/resources that are being used by the arkit. And he should foresee this and only reproduce the video only when he has the necessary resources.

Here is a code, I try to use as many resources as possible to try to reproduce what happens using the arkit, if u open on the first time (at least that's what happens to me) it shows a white screen and only then shows the video.

and it happens even worse with the arkit, but the video runs smoothly. attention I am not using both at the same time. I have the arkit running and only after the arkit is finished, I use the video plugin. I even giving a delay in the video plugin and sometimes still shows the white window


Code:
`class _ImageDetectionPageState extends State<MyHomePage> {
  Future<String> firstAsync() async {
    //await Future<String>.delayed(const Duration(seconds: 2));
    for(int i = 0; i<9000; i++){
      print("1111111111111111111");
    }
    return "First!";
  }

  Future<String> secondAsync() async {
    await Future<String>.delayed(const Duration(seconds: 1));
    for(int i = 0; i<9000; i++){
      print("222222222222222222222");
    }
    return "Second!";
  }

  Future<String> thirdAsync() async {
    await Future<String>.delayed(const Duration(seconds: 2));
    for(int i = 0; i<9000; i++){
      print("333333333333333");
    }
    return "Third!";
  }

  Future<String> fourthAsync() async {
    await Future<String>.delayed(const Duration(seconds: 3));
    for(int i = 0; i<10000; i++){
      print("4444444444");
    }
    return "Third!";
  }



  Future<void> cccccc() async{
    Future.delayed(
      Duration(seconds: 0),
      () => {
        Navigator.push<PlayerVideoAndPopPage>(
          context,
          MaterialPageRoute<PlayerVideoAndPopPage>(
            builder: (BuildContext context) => PlayerVideoAndPopPage(),
          ),
        ),
      },
    );
  }

  Widget build(BuildContext context) {
    return DefaultTabController(
      length: 3,
      child: Scaffold(
        key: const ValueKey<String>('home_page'),
        appBar: AppBar(
          title: const Text('Video player example'),
          actions: <Widget>[
            IconButton(
              key: const ValueKey<String>('push_tab'),
              icon: const Icon(Icons.navigation),
              onPressed: () async {
                await cccccc();
                var f = firstAsync();
                var s = secondAsync();
                var t = thirdAsync();
                var v = fourthAsync();
                print(await f);
                print(await s);
                print(await t);
                print(await v);
                print('done');
              },
            )
          ],
        ),
      ),
    );
  }
}`

**Comment by TahaTesser** [comment]
Hi @rubenteste 
I tried to replicate the issue based on your code snippet, here I created a minimal runnable code sample, video plays fine. 


<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:video_player/video_player.dart';

void main() => runApp(MyApp());

// ignore: public_member_api_docs
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

// ignore: public_member_api_docs
class Home extends StatefulWidget {

  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  Future<String> firstAsync() async {
    //await Future<String>.delayed(const Duration(seconds: 2));
    for(int i = 0; i<9000; i++){
      print("1111111111111111111");
    }
    return "First!";
  }

  Future<String> secondAsync() async {
    await Future<String>.delayed(const Duration(seconds: 1));
    for(int i = 0; i<9000; i++){
      print("222222222222222222222");
    }
    return "Second!";
  }

  Future<String> thirdAsync() async {
    await Future<String>.delayed(const Duration(seconds: 2));
    for(int i = 0; i<9000; i++){
      print("333333333333333");
    }
    return "Third!";
  }

  Future<String> fourthAsync() async {
    await Future<String>.delayed(const Duration(seconds: 3));
    for(int i = 0; i<10000; i++){
      print("4444444444");
    }
    return "Third!";
  }



  Future<void> cccccc() async{
    Future.delayed(
      Duration(seconds: 0),
      () => {
        Navigator.push<PlayerVideoAndPopPage>(
          context,
          MaterialPageRoute<PlayerVideoAndPopPage>(
            builder: (BuildContext context) => PlayerVideoAndPopPage(),
          ),
        ),
      },
    );
  }



  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      length: 3,
      child: Scaffold(
        key: const ValueKey<String>('home_page'),
        appBar: AppBar(
          title: const Text('Video player example'),
          actions: <Widget>[
            IconButton(
              key: const ValueKey<String>('push_tab'),
              icon: const Icon(Icons.navigation),
              onPressed: () async {
                await cccccc();
                var f = firstAsync();
                var s = secondAsync();
                var t = thirdAsync();
                var v = fourthAsync();
                print(await f);
                print(await s);
                print(await t);
                print(await v);
                print('done');
              },
            )
          ],
        ),
      ),
    );
  }
}


class PlayerVideoAndPopPage extends StatefulWidget {
  @override
  _PlayerVideoAndPopPageState createState() => _PlayerVideoAndPopPageState();
}

class _PlayerVideoAndPopPageState extends State<PlayerVideoAndPopPage> {
 late  VideoPlayerController _controller;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network(
        'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4')
      ..initialize().then((_) {
        // Ensure the first frame is shown after the video is initialized, even before the play button has been pressed.
        setState(() {});
      });
          _controller.setLooping(true);

    _controller.play();
  }


    @override
  void dispose() {
    super.dispose();
    _controller.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(
          child: _controller.value.isInitialized
              ? AspectRatio(
                  aspectRatio: _controller.value.aspectRatio,
                  child: VideoPlayer(_controller),
                )
              : Container(),
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: () {
            setState(() {
              _controller.value.isPlaying
                  ? _controller.pause()
                  : _controller.play();
            });
          },
          child: Icon(
            _controller.value.isPlaying ? Icons.pause : Icons.play_arrow,
          ),
        ),
      );
  }
}
``` 

</details>

If it is about resources, you can use third-party packages but please provide a working code sample with steps to reproduce the issue, instead of a code snippet. Feel free to edit my code sample and add arkit code, provide the reproducible code sample
Thank you

**Comment by rubenteste** [comment]
Hi
So I used your code, changed the part of the video a bit so I didn't have the play / pause button and added the arkit plugin.

but because I use an image to anchor it is necessary to open the project in Xcode to create a folder with all the images to be used as anchors 
![Captura de ecrã 2021-05-04, às 12 18 35](https://user-images.githubusercontent.com/75133022/116996242-06f92400-acd3-11eb-86e7-daf457cc97d4.png)

![Captura de ecrã 2021-05-04, às 12 19 20](https://user-images.githubusercontent.com/75133022/116996256-0bbdd800-acd3-11eb-9189-b4e35ac78782.png)


That's why the whole project is there because you just need to open it, you already have the folder with the necessary image. 
try testing the application 9/10 times until the video appears. then go back until Tap option appears again and point to the world on wiki, sooner or later the video will have problems

tell me something if you can't get the code running in one way or another

code sample:
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:video_player/video_player.dart';

import 'package:arkit_plugin/arkit_plugin.dart';
import 'package:vector_math/vector_math_64.dart' as vector;

void main() => runApp(MaterialApp(home: MyApp()));

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final samples = [
      Sample(
        'Tap',
        'Sphere which handles tap event.',
        Icons.touch_app,
            () => Navigator.of(context)
            .push<void>(MaterialPageRoute(builder: (c) => TapPage())),
      ),
    ];
    return Scaffold(
      appBar: AppBar(
        title: const Text('ARKit Demo'),
      ),
      body:
      ListView(children: samples.map((s) => SampleItem(item: s)).toList()),
    );
  }
}

class SampleItem extends StatelessWidget {
  const SampleItem({Key key, this.item}) : super(key: key);
  final Sample item;

  @override
  Widget build(BuildContext context) {
    return Card(
      child: InkWell(
        onTap: () => item.onTap(),
        child: ListTile(
          leading: Icon(item.icon),
          title: Text(
            item.title,
            style: Theme.of(context).textTheme.subtitle1,
          ),
          subtitle: Text(
            item.description,
            style: Theme.of(context).textTheme.subtitle2,
          ),
        ),
      ),
    );
  }
}

class Sample {
  const Sample(this.title, this.description, this.icon, this.onTap);
  final String title;
  final String description;
  final IconData icon;
  final Function onTap;
}

class TapPage extends StatefulWidget {
  @override
  _ImageDetectionPageState createState() => _ImageDetectionPageState();

}

class _ImageDetectionPageState extends State<TapPage> {
  ARKitController arkitController;
  Timer timer;
  bool anchorWasFound = false;

  @override
  void dispose() {
    timer?.cancel();
    arkitController?.dispose();
    super.dispose();
  }

  Future<void> cccccc() async{
    Future.delayed(
      Duration(seconds: 0),
          () => {
        Navigator.push<PlayerVideoAndPopPage>(
          context,
          MaterialPageRoute<PlayerVideoAndPopPage>(
            builder: (BuildContext context) => PlayerVideoAndPopPage(),
          ),
        ).then((value){
          setState(() {
            anchorWasFound = false;
          });
        }),
      },
    );
  }

  @override

  Widget build(BuildContext context) => Scaffold(
    appBar: AppBar(title: const Text('Image Detection Sample')),
    body: Container(
      child: Stack(
        fit: StackFit.expand,
        children: [
          ARKitSceneView(
            detectionImagesGroupName: 'AR Resources',
            onARKitViewCreated: onARKitViewCreated,
          ),
          anchorWasFound
              ? Container()
              : Padding(
            padding: const EdgeInsets.all(8.0),
            child: Text(
              'Point the camera at the earth image from the article about Earth on Wikipedia.',
              style:
                Theme.of(context).textTheme.headline5.copyWith(color: Colors.white),
            ),
          ),
        ],
      ),
    ),
  );


  void onARKitViewCreated(ARKitController arkitController) {
    this.arkitController = arkitController;
    this.arkitController.onAddNodeForAnchor = onAnchorWasFound;
    print("llllllllll");
    print("llllllllll");
    print("llllllllll");
  }

  void onAnchorWasFound(ARKitAnchor anchor) {
    if (anchor is ARKitImageAnchor) {
      cccccc();
    }
  }
}

class PlayerVideoAndPopPage extends StatefulWidget {
  @override
  _PlayerVideoAndPopPageState createState() => _PlayerVideoAndPopPageState();
}

class _PlayerVideoAndPopPageState extends State<PlayerVideoAndPopPage> {
  VideoPlayerController _controller;
  bool startedPlaying = false;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4');
    _controller.addListener(() {
      if (startedPlaying && !_controller.value.isPlaying) {
        Navigator.pop(context);

      }
    });
  }

  Future<bool> started() async {
    await _controller.initialize();
    await _controller.play();
    startedPlaying = true;
    return true;
  }

  @override
  void dispose() {
    super.dispose();
    _controller.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: FutureBuilder<bool>(
          future: started(),
          builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
            if (snapshot.data == true) {
              return AspectRatio(
                aspectRatio: _controller.value.aspectRatio,
                child: VideoPlayer(_controller),
              );
            } else {
              return const Text("");
            }
          },
        ),
      ),
    );
  }
}

Full project :
https://we.tl/t-kvnduq9mTj

**Comment by TahaTesser** [comment]
Hi @rubenteste 
Thanks for the code, I downloaded the full project, ran it on my physical iPhone XR 20 times, and each time the video frame is loaded, video and audio are played, multiple frames are played.  Never saw a white or blank screen or white screen in the video player itself, I am running in debug mode.

Repeated these steps 20 times
Tap list tile on the home, scan [Earth image from wiki](https://upload.wikimedia.org/wikipedia/commons/9/97/The_Earth_seen_from_Apollo_17.jpg) and it navigates to video page


<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3 20E232 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 1d9032c7e1 (5 days ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      14.5
    • macOS (desktop)        • macos                     • darwin-x64     •
      macOS 11.3 20E232 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 90.0.4430.93

• No issues found!
```
		
</details>

**Comment by rubenteste** [comment]
Hi


i'm using an iphone 7 maybe is because of that, yours is newer and has a higher processing power ...
I haven't tested it in debug mode only in normal mode, can you try to run in normal mode without debugging? and if you have an older iphone can test on that?

**Comment by TahaTesser** [comment]
Hi @rubenteste 
I have an iPhone 5s with `iOS 12.5.2`, its processor doesn't support ARKit so I tried the earlier simpler example https://github.com/flutter/flutter/issues/81480#issuecomment-831729698 for around 20 times and it also plays the video every time, although, it has a slight delay in loading the first frame but every frame loads and there is no freeze or a blank screen. Even the device got hot, it still played the video

### Preview (iPhone 5s, repeat these steps 20 times)

https://user-images.githubusercontent.com/48603081/117258819-0b9a1580-ae56-11eb-9164-e99023e72fb7.MP4




<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3 20E232 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 1d9032c7e1 (6 days ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • e534bcc711c9a011aa892eb51cdb60f40b3acddb • ios
      • iOS 12.5.2
    • macOS (desktop)        • macos                                    •
      darwin-x64     • macOS 11.3 20E232 darwin-x64
    • Chrome (web)           • chrome                                   •
      web-javascript • Google Chrome 90.0.4430.93

• No issues found!
```
		
</details>

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
